internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -262,6 +271,32 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 				}, false),
 		},
 
+		"nginx_app_protect": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"web_application_firewall_settings": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"activation_state": {
+									Type:     pluginsdk.TypeString,
+									Optional: true,
```

Comment:
is the activation state not a required input?

Reply:
what ends up happening if you do not specify the activation state?

Reply:
The activation state is optional in the rest-api, but it makes sense to make it required. If no activation state is specified then we default to Disabled